src/styles/_loops.scss

```diff
@@ -0,0 +1,5 @@
+@for $num from 1 through 6 {
+  .blocks__block--#{$num} {
```

Comment:
You could do it with nth-child, and then each block wouldn't need a special class